File: android/foxui/src/main/java/io/gitee/stevenzack/foxui/FObject/Service/FService.java
package io.gitee.stevenzack.foxui.FObject.Service;

import android.content.Context;
import android.content.Intent;

import io.gitee.stevenzack.foxui.CoreService;
import io.gitee.stevenzack.foxui.FObject.FObject;
import io.gitee.stevenzack.foxui.FoxActivity;
import io.gitee.stevenzack.foxui.R;

public class FService extends FObject {
    public FService(FoxActivity foxActivity) {
        parentController = foxActivity;
    }

    public static String getAttrStaticly(String attr) {
        return "";
    }

    public static void setAttr(Context context, String attr, final String value) {
        switch (attr) {
            case "OnCreate":
                CoreService.onCreate = value;
                break;
            case "QuitButton":
                CoreService.quitButton = value;
                break;
            case "OnQuit":
                CoreService.onQuitClick =value;
                break;
            case "Title":
                CoreService.ntf_title=value;
                if (CoreService.notification!=null){
                    CoreService.getInstance().handler.post(new Runnable() {
                        @Override
                        public void run() {
                            CoreService.notification.contentView.setTextViewText(R.id.ntf_title,value);
                            CoreService.getInstance().startForeground(1,CoreService.notification);
                        }
                    });
                }
                break;
            case "SubTitle":
                CoreService.ntf_subtitle = value;
                if (CoreService.notification != null) {
                    CoreService.getInstance().handler.post(new Runnable() {
                        @Override
                        public void run() {
                            CoreService.notification.contentView.setTextViewText(R.id.ntf_subtitle,value);
                            CoreService.getInstance().startForeground(1,CoreService.notification);
                        }
                    });
                }
                break;
            case "Show":
                Intent intent = new Intent(context, CoreService.class);
                context.startService(intent);
                break;
            case "FinishAllActivity":
                CoreService.getInstance().finishAllActivity();
                break;
            case "KillAll":
                CoreService.killAll();
                break;
            case "Stop":
                CoreService.getInstance().stopService();
                break;
        }
    }
}

File: fcore/widget/fservice/fservice.go
package fservice

import "github.com/StevenZack/fox/fcore"

type FService struct {
	a fcore.IActivity
}

func StartService(a fcore.IActivity, onCreate func()) *FService {
	fnId:=fcore.NewToken()
	fcore.EventMap.Set(fnId, func(activity fcore.IActivity, s string, s2 string, s3 string) string {
		onCreate()
		return ""
	})
	f:=&FService{a:a}
	f.a.SetAttr("Service","OnCreate",fnId,"")
	return f
}
func New(a fcore.IActivity) *FService {
	return &FService{a:a}
}
func (f *FService) QuitButton(title string) *FService {
	f.a.SetAttr("Service","QuitButton",title,"")
	return f
}
func (f *FService) Title(title string) *FService {
	f.a.SetAttr("Service","Title",title,"")
	return f
}
func (f *FService) SubTitle(sub string) *FService {
	f.a.SetAttr("Service","SubTitle",sub,"")
	return f
}
func (f *FService) OnCreate(fn func()) *FService {
	fnId:=fcore.NewToken()
	fcore.EventMap.Set(fnId, func(activity fcore.IActivity, s string, s2 string, s3 string) string {
		fn()
		return ""
	})
	f.a.SetAttr("Service","OnCreate",fnId,"")
	return f
}
func (f *FService) OnQuit(fn func()) *FService {
	fnId:=fcore.NewToken()
	fcore.EventMap.Set(fnId, func(activity fcore.IActivity, s string, s2 string, s3 string) string {
		fn()
		return ""
	})
	f.a.SetAttr("Service","OnQuit",fnId,"")
	return f
}
func (f *FService) Show() *FService {
	f.a.SetAttr("Service","Show","","")
	return f
}
func (f *FService) FinishAllActivity() *FService {
	f.a.SetAttr("Service","FinishAllActivity","","")
	return f
}
func (f *FService) KillAll() *FService {
	f.a.SetAttr("Service","KillAll","","")
	return f
}
func (f *FService) Stop() *FService {
	f.a.SetAttr("Service","Stop","","")
	return f
}